File: Own practices/java/find_2nd_num.java
public class find_2nd_num {

    public static int secondLargest(int[] arr) {

        if (arr.length < 2)
            return -1;

        int largest = Integer.MIN_VALUE;
        int second = Integer.MIN_VALUE;

        for (int num : arr) {

            if (num > largest) {
                second = largest;
                largest = num;
            } else if (num > second && num != largest) {
                second = num;
            }
        }

        if (second == Integer.MIN_VALUE)
            return -1;

        return second;
    }

    public static void main(String[] args) {

        int[] arr = {10, 20, 5, 20, 8, 18, 15};

        System.out.println(secondLargest(arr));
    }
}

File: Own practices/py/find_2nd_num.py
#to find 2nd largest number

def second_largest(arr):
    if len(arr) < 2:
        return -1

    largest = float('-inf')
    second = float('-inf')

    for num in arr:
        if num > largest:
            second = largest
            largest = num
        elif num > second and num != largest:
            second = num

    if second == float('-inf'):
        return -1

    return second


arr = [10, 20, 5, 20, 8, 15]
print(second_largest(arr))